tests/secc/states/test_iso15118_2_states.py

```diff
@@ -77,3 +79,110 @@ async def test_authorization_next_state_on_authorization_request(
+    async def test_charge_parameter_discovery_res_v2g2_304(self):
+        # V2G2-304: If departure time was not provided, then sum of time intervals
+        # in PMaxSchedule shall be greater than or equal to 24 hours.
+        twenty_four_hours_in_seconds = 86400
+        charge_parameter_discovery = ChargeParameterDiscovery(self.comm_session)
+        charge_parameter_discovery.process_message(
+            message=get_charge_parameter_discovery_req_message_no_departure_time()
+        )
+        assert (
+            charge_parameter_discovery.next_msg.body.charge_parameter_discovery_res
+            is not None
+        )
+        charge_parameter_discovery_res = (
+            charge_parameter_discovery.next_msg.body.charge_parameter_discovery_res
+        )
+        assert charge_parameter_discovery_res.sa_schedule_list is not None
+        schedule_tuples = (
+            charge_parameter_discovery_res.sa_schedule_list.schedule_tuples
+        )
+
+        for schedule_tuple in schedule_tuples:
+            schedule_duration = 0
+            if schedule_tuple.p_max_schedule.schedule_entries is not None:
+                start_time = schedule_tuple.p_max_schedule.schedule_entries[
+                    0
+                ].time_interval.start
+
+            for entry in schedule_tuple.p_max_schedule.schedule_entries:
+                schedule_duration += entry.time_interval.start - start_time
+                if entry.time_interval.duration is not None:
+                    schedule_duration += entry.time_interval.duration
+                start_time = entry.time_interval.start
+            assert schedule_duration >= twenty_four_hours_in_seconds
+
+    async def test_charge_parameter_discovery_res_v2g2_761(self):
+        # V2G2-761: If departure time was not provided, then SECC shall assume
+        # that the EV intends to start charging without any delay
+        charge_parameter_discovery = ChargeParameterDiscovery(self.comm_session)
+        charge_parameter_discovery.process_message(
+            message=get_charge_parameter_discovery_req_message_no_departure_time()
+        )
+        assert (
+            charge_parameter_discovery.next_msg.body.charge_parameter_discovery_res
+            is not None
+        )
+        charge_parameter_discovery_res = (
+            charge_parameter_discovery.next_msg.body.charge_parameter_discovery_res
+        )
+
+        assert charge_parameter_discovery_res.sa_schedule_list is not None
+        sa_schedule_tuples = (
+            charge_parameter_discovery_res.sa_schedule_list.schedule_tuples
+        )
+
+        for schedule_tuples in sa_schedule_tuples:
+            assert schedule_tuples.p_max_schedule is not None
+            found_entry_indicating_start_without_delay = False
+            duration_indicating_immediate_start = 30
+            for entry in schedule_tuples.p_max_schedule.schedule_entries:
+                if entry.time_interval.start < duration_indicating_immediate_start:
```

Comment:
where does this 30 secs value comes from? is it a recommendation?
I think the start entry should have a start value = 0

Reply:
I was doubtful about this - added 30 as a rough indicator of immediate start - have set it to 0.